src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/service/MessageNotifier.kt

```diff
@@ -35,7 +55,34 @@ class MessageNotifier(
         .build()
 
     val publishResult = topic.publish(eventType = "commonplatform.case.received", event = objectMapper.writeValueAsString(hearingEvent), attributes = mapOf("messageType" to messageTypeValue, "hearingEventType" to hearingEventTypeValue), messageGroupId = MESSAGE_GROUP_ID)
-    log.info("Published message with message Id {}", publishResult.messageId())
+
+    return publishResult.messageId()
+  }
+
+  fun publishLargeMessage(hearingEvent: HearingEvent, hearingEventType: HearingEventType): String? {
+    val snsExtendedAsyncClientConfiguration: SNSExtendedAsyncClientConfiguration = SNSExtendedAsyncClientConfiguration()
+      .withPayloadSupportEnabled(amazonS3AsyncClient, bucketName)
+
+    val snsExtendedClient = AmazonSNSExtendedAsyncClient(
+      topic.snsClient,
+      snsExtendedAsyncClientConfiguration,
+    )
+    // Publish message via SNS with storage in S3
+    val publishResult = snsExtendedClient.publish(
+      PublishRequest.builder().topicArn(topic.arn).messageAttributes(
+        mapOf(
+          "messageType" to MessageAttributeValue.builder().dataType("String").stringValue(MESSAGE_TYPE).build(),
```

Comment:
Any chance we could use a different messageType here?
How about `commonplatform.large.case.received`?
Would make processing a bit simpler